### PR TITLE
[TRANSMITTER] [EMAIL] use BCC field

### PR DESCRIPTION
**Pull request by jerome-poisson**
added cc and bcc handling in send_email.
The new recipients_bcc is used in email transmitter.
Improved field parsing so spaces can be used around emails

### CODE REVIEW

File: superdesk/publish/transmitters/email.py [hunk 1]
```diff
@@ -37,11 +37,13 @@ def _transmit(self, queue_item, subscriber):
...
 
-            if not config.get('recipients'):
+            if not config.get('recipients') and not config.get('recipients_bcc'):
                 raise PublishEmailError.recipientNotFoundError(LookupError('recipient field not found!'))
 
             admins = app.config['ADMINS']
-            recipients = config.get('recipients').rstrip(';').split(';')
+            recipients = [r.strip() for r in config.get('recipients', '').split(';') if r.strip()]
+            bcc = [r.strip() for r in config.get('recipients_bcc', '').split(';') if r.strip()]
+            assert recipients or bcc
```
Comment (naspeh) [inline superdesk/publish/transmitters/email.py]: Probably `raise PublishEmailError.recipientNotFoundError` would be better here

Comment (jerome-poisson) [inline superdesk/publish/transmitters/email.py]: indeed, fixed

File: superdesk/publish/transmitters/email.py [hunk 2]
```diff
@@ -37,11 +37,12 @@ def _transmit(self, queue_item, subscriber):
...
 
-            if not config.get('recipients'):
-                raise PublishEmailError.recipientNotFoundError(LookupError('recipient field not found!'))
-
             admins = app.config['ADMINS']
-            recipients = config.get('recipients').rstrip(';').split(';')
+            recipients = [r.strip() for r in config.get('recipients', '').split(';') if r.strip()]
+            bcc = [r.strip() for r in config.get('recipients_bcc', '').split(';') if r.strip()]
+            if not recipients and not bcc:
+                raise PublishEmailError.recipientNotFoundError(LookupError('recipient or bcc field not found!'))
```
Comment (naspeh) [inline superdesk/publish/transmitters/email.py]: probably message should be `recipient and bcc are empty` :)

**Comment by naspeh** [comment]
```
++ flake8 --jobs=1
./superdesk/publish/transmitters/email.py:47:13: E303 too many blank lines (2)
```

File: superdesk/publish/transmitters/email.py [hunk 3]
```diff
@@ -37,11 +37,11 @@ def _transmit(self, queue_item, subscriber):
...
 
-            if not config.get('recipients'):
-                raise PublishEmailError.recipientNotFoundError(LookupError('recipient field not found!'))
-
             admins = app.config['ADMINS']
-            recipients = config.get('recipients').rstrip(';').split(';')
+            recipients = [r.strip() for r in config.get('recipients', '').split(';') if r.strip()]
+            bcc = [r.strip() for r in config.get('recipients_bcc', '').split(';') if r.strip()]
+            if not recipients and not bcc:
+                raise PublishEmailError.recipientNotFoundError(LookupError('recipient and bcc fieldis are empty!'))
```
Comment (naspeh) [inline superdesk/publish/transmitters/email.py]: `s/fieldis/fields` typo here

**Pull request merged**